crates/curp/src/server/raw_curp/mod.rs

```diff
@@ -101,6 +101,8 @@ pub struct RawCurp<C: Command, RC: RoleChange> {
     cst: Mutex<CandidateState<C>>,
     /// Curp logs
     log: RwLock<Log<C>>,
+    /// Contexts of fallback log entries
+    fallback_contexts: DashMap<LogIndex, FallbackContext<C>>,
```

Comment:
Please use a `RwLock<HashMap>` for this, as we don't need to concurrently modify the map.